language/deepseek-r1/run_mlperf.py

```diff
@@ -0,0 +1,486 @@
+        max_length: Maximum sequence length
+        use_chat_template: Whether to use chat template if available
+        backend_name: Name of the backend being used
+
+    Returns:
+        Tuple of (tokenized_prompts, processed_strings)
+    """
+    logger.info(f"Loading tokenizer: {tokenizer_name}")
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+    # Automatically disable chat template for SGLang backend (like run_eval.py)
+    if backend_name == 'sglang':
+        use_chat_template = False
+        logger.info("[SGLang] Automatically disabling chat template for SGLang backend")
+
+    tokenized = []
+    processed_strings = []
+
+    for prompt in prompts:
+        # Check if tokenizer has chat template and we want to use it
+        if use_chat_template and hasattr(tokenizer, 'apply_chat_template'):
+            tokens = tokenizer.apply_chat_template(
+                [{"role": "user", "content": prompt}],
+                add_generation_prompt=True,
+                max_length=max_length,
+                truncation=True
+            )
+            processed_string = tokenizer.decode(tokens, skip_special_tokens=False)
+        else:
+            tokens = tokenizer.encode(
+                prompt,
+                truncation=True,
+                max_length=max_length
+            )
+            processed_string = prompt
+
+        tokenized.append(tokens)
+        processed_strings.append(processed_string)
+
+    return tokenized, processed_strings
+
+
+def configure_loadgen(scenario: str,
+                     accuracy_mode: bool,
+                     mlperf_conf: Optional[str] = None,
+                     user_conf: Optional[str] = None,
+                     log_dir: Optional[str] = None,
+                     model_name: str = "deepseek-r1") -> lg.TestSettings:
+    """Configure LoadGen test settings.
+
+    Args:
+        scenario: MLPerf scenario ("offline" or "server")
+        accuracy_mode: Whether to run in accuracy mode
+        mlperf_conf: Path to MLPerf config file
+        user_conf: Path to user config file
+        log_dir: Directory for logs
+        model_name: Model name for configuration (default: deepseek-r1)
+
+    Returns:
+        LoadGen TestSettings
+    """
+    settings = lg.TestSettings()
+
+    # Set scenario
+    if scenario.lower() == "offline":
+        settings.scenario = lg.TestScenario.Offline
+    elif scenario.lower() == "server":
+        settings.scenario = lg.TestScenario.Server
+    else:
+        raise ValueError(f"Unknown scenario: {scenario}")
+
+    # Set mode
+    if accuracy_mode:
+        settings.mode = lg.TestMode.AccuracyOnly
+    else:
+        settings.mode = lg.TestMode.PerformanceOnly
+
+    # Load configurations if files exist
+    if mlperf_conf and Path(mlperf_conf).exists():
+        settings.FromConfig(mlperf_conf, model_name, scenario)
+    if user_conf and Path(user_conf).exists():
+        settings.FromConfig(user_conf, model_name, scenario)
```

Comment:
@pgmpablo157321 the mlperf.conf doesn't have deepseek for now, so Viraat had to hack all of them in user.conf right now. Please open a separate PR to add the official DS support